support/android/apk/build.gradle

```diff
@@ -79,3 +81,22 @@ ext.getSigningKeyInfo = { ->
         return null
     }
 }
+
+ext {
+  // Generate unique version code based on the build date and time to support nightly
+  // builds.
+  //
+  // The version scheme is currently: yyDDDHHmm
+  // where
+  // yy is the double digit year (e.g 24 for 2024)
+  // DDD is the 3 digit (zero padded) day of the year (e.g 257 for September 13)
+  // HHmm is the time of the day (e.g 1720 for 5.20 pm)
+  //
+  // TODO: check if this interferes with caching of local builds and add option to use
+  // a static version.
+  def today = new Date()
+  def year = new SimpleDateFormat("yy").format(today)
+  def day = String.format("%03d", (new SimpleDateFormat("D").format(today) as int))
+  def time = new SimpleDateFormat("HHmm").format(today)
+  generatedVersionCode = (year + day + time) as int
+}
```

Comment:
Why don't we do four digits for the year and use two digits for the version inside the day? So for instance for today: 2024091600, where the final two digits are for releases in the same day

Reply:
I went with the current scheme to minimize the number of digits to keep the available version space large for future expansion (using potentially different schemes) and also to not rely on external state to distinguish multiple builds (like a counter). However, the cost of changing the scheme in backwards-incompatible manner is not too high, the user just needs to uninstall and reinstall the app, so I'll change the scheme as you've suggested (with 00 for the last digits for now).

Reply:
I've addressed this in [85be322](https://github.com/servo/servo/pull/33435/commits/85be322389ce71436e44f3a86f6357e9609e01f6)